pulp_container/tests/functional/api/test_push_content.py

```diff
@@ -6,49 +6,186 @@
+                "container.view_containerdistribution",
+                "container.add_containernamespace",
+            ]
+        )
+        cls.user_collaborator = gen_user(
+            [
+                "container.add_containerdistribution",
+                "container.view_containerdistribution",
+                "container.manage_namespace_distributions",
+                "container.view_containerpushrepository",
+            ]
+        )
+        cls.user_reader = gen_user(["container.view_containerpushrepository"])
+        cls.user_helpless = gen_user([])
 
-class PushContentTestCase(unittest.TestCase):
-    """Verify whether images can be pushed to pulp."""
+        cls._pull("centos:7")
+        cls._pull(f"{DOCKERHUB_PULP_FIXTURE_1}:manifest_a")
+        cls._pull(f"{DOCKERHUB_PULP_FIXTURE_1}:manifest_b")
 
-    def test_push_using_registry_client(self):
-        """Test push with official registry client"""
-        registry.raise_if_unsupported(unittest.SkipTest, "Test requires podman/docker")
-        registry_name = urlparse(cfg.get_base_url()).netloc
-        local_url = "/".join([registry_name, "foo/bar:1.0"])
-        # Be sure to not being logged in
-        try:
-            registry.logout(registry_name)
-        except exceptions.CalledProcessError:
-            pass
-        # Pull an image with large blobs
-        registry.pull("centos:7")
+    @classmethod
+    def tearDownClass(cls):
+        """Delete api users."""
+        del_user(cls.user_creator)
+        del_user(cls.user_collaborator)
+        del_user(cls.user_reader)
+        del_user(cls.user_helpless)
+
+    @classmethod
+    def _pull(cls, image_path, user=None):
+        """
+        Pull using specified user.
+
+        Ensure we login with a user if specified and logout after the pull.
+
+        If user is not specified, ensure that no other user is logged in and pull is performed
+        anonymously.
+        """
+        if user:
+            cls.registry.login("-u", user["username"], "-p", user["password"], cls.registry_name)
+        else:
+            # Ensure logout
+            try:
+                cls.registry.logout(cls.registry_name)
+            except exceptions.CalledProcessError:
+                pass
+
+        cls.registry.pull(image_path)
+
+        if user:
+            cls.registry.logout(cls.registry_name)
+
+    @classmethod
+    def _push(cls, image_path, local_url, user):
+        """
+        Tag and push an image to Pulp registry using specified user.
+
+        Ensure we login with a specified user and logout after the push.
+        A local tag is removed for cleanup purposes.
+        """
         # Tag it to registry under test
-        registry.tag("centos:7", local_url)
+        cls.registry.tag(image_path, local_url)
+        # Log in
+        cls.registry.login("-u", user["username"], "-p", user["password"], cls.registry_name)
+        try:
+            # Push successfully
```

Comment:
This comment should be removed, IMHO. Because this function can just as well test an unsuccessful push operation.